### PR TITLE
Little ganga import

### PR DESCRIPTION
**Pull request by drmarkwslater**
Added a new `ganga` module that does the usual bootstrap (except the monitoring) and then imports all the GPI. This allows:

```
from ganga import *
j = Job()
j.submit()
```

### CODE REVIEW

**Comment by milliams** [comment]
add to whitelist

File: python/ganga/__init__.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+            logger = Ganga.Utility.logging.getLogger()
+            logger.error("Object %s DOES NOT have the _name parameter set" % (str(obj)))
+            return ""
+    else:
+        if hasattr(obj, '_name'):
+            return obj._name
+        else:
+            logger = Ganga.Utility.logging.getLogger()
+            logger.error("Object %s DOES NOT have the %s or _name parameter set" % (str(obj), str(proxyRef)))
+            return ""
```
Comment (milliams) [inline python/ganga/__init__.py]: `return stripProxy(obj)._name` should be sufficient for any proxies or `GangaObject`s. Likewise for the `_category` below.

File: python/ganga/__init__.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        for c in allPlugins.allCategories():
+            d[c] = allPlugins.allCategories()[c].keys()
+        return d
+
+
+def list_plugins(category):
+    """List all plugins in a given category, OBSOLETE: use plugins(category)"""
+    logger.warning('This function is deprecated, use plugins("%s") instead', category)
+    from Ganga.Utility.Plugin import allPlugins
+    return allPlugins.allClasses(category).keys()
```
Comment (milliams) [inline python/ganga/__init__.py]: Let's not add obsolete functions into the new API.

**Comment by drmarkwslater** [comment]
@milliams Before you go to far - this is a cut and paste job (mostly) of the bootstrap code to make sure it everything got in. It was not obvious most of the time how some components were included so as a start I took everything. I do intend to work through this much more carefully but if I'd prefer to get it in for 6.1.15 in *some* form. I'll leave it up to you as release manager though - I can start going through it properly now and leave it until 6.1.16 or we can put it in now and fix later - just let me know!

**Comment by milliams** [comment]
@drmarkwslater I wasn't about to start critiquing the details at all I thought that `typename` was something you'd added but if it's a copy then leave it for now.

It's probably best to tear out the obsolete functions asap though (although I won't hold up merging for it).

That aside, I'm happy to merge so that we can start testing this.

**Comment by drmarkwslater** [comment]
@milliams Should be better - and also now works with the latest changes in develop :)

**Comment by milliams** [comment]
:+1:

**Pull request merged**